Tests/Exporter/ExporterTest.php

```diff
@@ -12,21 +12,64 @@
 namespace Sonata\CoreBundle\Tests\Exporter;
 
 use Exporter\Source\ArraySourceIterator;
+use Exporter\Writer\CsvWriter;
+use Exporter\Writer\JsonWriter;
+use Exporter\Writer\XlsWriter;
+use Exporter\Writer\XmlWriter;
 use Sonata\CoreBundle\Exporter\Exporter;
 
 class ExporterTest extends \PHPUnit_Framework_TestCase
 {
-    /**
-     * @expectedException RuntimeException
-     */
     public function testFilter()
     {
+        $this->setExpectedException('RuntimeException', 'Invalid "foo" format');
         $source = $this->getMock('Exporter\Source\SourceIteratorInterface');
+        $writer = $this->getMock('Exporter\Writer\TypedWriterInterface');
 
-        $exporter = new Exporter();
+        $exporter = new Exporter(array($writer));
         $exporter->getResponse('foo', 'foo', $source);
     }
 
+    public function testConstructorRejectsNonTypedWriters()
+    {
+        $this->setExpectedException(
+            PHP_VERSION_ID < 70000 ? 'PHPUnit_Framework_Error' : 'TypeError',
```

Comment:
It's a feature of phpunit, it converts errors to exception, so that you can assert that an error should occur. But do you mean I should except something less precise, like maybe `\Exception` ?